app/uk/gov/hmrc/gform/gform/AcknowledgementController.scala

```diff
@@ -87,49 +88,52 @@ class AcknowledgementController(
     submissionReference: Option[TextExpression],
     retrievals: Retrievals,
     hashedValue: String,
+    formTemplate: FormTemplate,
     data: Map[FormComponentId, Seq[String]]
-  ): String = {
+  )(implicit hc: HeaderCarrier): Future[String] = {
```

Comment:
At the return from this method we are unnecessary throwing away information we are returning `org.jsoup.nodes.Document`. We may hold this information little longer and convert to `String` when necessary later.

Reply:
Yes.  Further, we should not use jsoup like that at all.  HTML should be generated correctly from views, not generated and then edited.